Microsoft.Azure.Cosmos/src/Resource/CosmosExceptions/CosmosExceptionFactory.cs

```diff
@@ -20,7 +20,16 @@ internal static class CosmosExceptionFactory
             {
                 foreach (string key in dce.Headers)
                 {
-                    headers.Add(key, dce.Headers[key]);
+                    string value = dce.Headers[key];
+                    if (value == null)
+                    {
+                        throw new ArgumentNullException(
+                            message: $"{nameof(key)}: {key}; {nameof(value)}: {dce.Headers[key] ?? "null"};",
```

Comment:
nit: lookup can be removed?

```suggestion
                            message: $"{nameof(key)}: {key}; {nameof(value)}: {value ?? "null"};",
```